src/WebJobs.Script/Workers/Profiles/WorkerProfileManager.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.WebJobs.Script.Workers.Profiles;
+using Microsoft.Azure.WebJobs.Script.Workers.Rpc;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers
+{
+    /// <summary>
+    /// The default profile manager that manages profiles from language workers
+    /// </summary>
+    internal class WorkerProfileManager : IWorkerProfileManager
+    {
+        private readonly ILogger _logger;
+        private readonly IEnumerable<IWorkerProfileConditionProvider> _conditionProviders;
+        private Dictionary<string, List<WorkerDescriptionProfile>> _profiles;
+        private string _activeProfile;
+
+        public WorkerProfileManager(ILogger<WorkerProfileManager> logger, IEnumerable<IWorkerProfileConditionProvider> conditionProviders)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _conditionProviders = conditionProviders ?? throw new ArgumentNullException(nameof(conditionProviders));
+            _profiles = new Dictionary<string, List<WorkerDescriptionProfile>>();
+            _activeProfile = string.Empty;
+        }
+
+        /// <inheritdoc />
+        public void SaveWorkerDescriptionProfiles(List<WorkerDescriptionProfile> workerDescriptionProfiles, string language)
+        {
+            _profiles[language] = workerDescriptionProfiles;
```

Comment:
While running locally, this code block as well as RpcWorkerConfigFactory section was called thrice. Each the they had the same worker config. Checking if it is safe to do this assignment.

Reply:
@fabiocav thoughts on this? I'm assuming this is expected, we could always check to see if the profile already exists in the dictionary before trying to assign / instead of overriding every time

Reply:
Was it on different instances of the `WorkerProfileManager`?

Reply:
@fabiocav  Is hashcode a good way to verify the instance? I tried `Console.WriteLine(this.GetHashCode());` after this line and get 3 outputs. The first hashcode is different. The second and third are the same.

Reply:
Without tracing the calls it would be difficult to say. For the instances, you can create and inspect the object ID while debugging, but I do expect a call in the parent scope and one in the inner scope (web and job host levels for workers initialized at each)